sharding/shard.go

```diff
@@ -0,0 +1,219 @@
+package sharding
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type shardBackend interface {
```

Comment:
I think this should be exported. I can envision that database implementations will want to have a test that ensures they meet the requirements of this interface.

For example, see how ethclient test works: https://github.com/ethereum/go-ethereum/blob/master/ethclient/ethclient_test.go